source/grid/grid_tools.cc

```diff
@@ -1573,6 +1573,16 @@ namespace GridTools
       return false;
     }
 
+    template <int dim, typename Cell>
+    inline
+    double calculate_cell_radius(const Cell &cell)
+    {
+      double res = 0.;
+      for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+        res = std::max(res, ( cell->vertex(v)).distance(cell->center()) );
```

Comment:
The above example makes sense to me. However, (after discussing with @davydden )this time we should change the associated function `calculate_cell_radius()` to `smallest_enclosing_sphere()` which returns a pair of `Point<dim>` corresponding to center of largest diagonal and `double` corresponding to the radius of the smallest enclosing sphere. I will push this function to this PR soon.